tools/generators/swift_debug_settings/src/SwiftDebugSettings.swift

```diff
@@ -12,16 +13,32 @@ struct SwiftDebugSettings {
 
         do {
             // First argument is executable name
-            var rawArguments = CommandLine.arguments.dropFirst()
+            let rawArguments = CommandLine.arguments.dropFirst()
+            
+            // Check for a params file
+            var arguments: ArraySlice<String>
+            if let arg = rawArguments.first, rawArguments.count == 1,
+               arg.starts(with: "@")
+            {
+                arguments = try await parseParamsFile(String(arg.dropFirst()))
+            } else {
+                arguments = ArraySlice(rawArguments)
+            }
```

Comment:
We have `parseAsRootSupportingParamsFile` for this, why can't we use that?

Reply:
It seemed more invasive to transition this to `ArgumentParser` and then declare a generic property to take in arbitrary key & file pairs to pass to the `Generator`. Similar logic applies to `TargetBuildSettings` too.

I'm happy to take that route though if you think it'll be cleaner.